lib/nerve/reporter/zookeeper.rb

```diff
@@ -62,15 +64,41 @@ def stop()
     end
 
     def report_up()
-      zk_save
+      if not @zk.connected?
+        log.error "nerve: error in reporting up on zk node #{@full_key}: loss connection"
+      else
+        begin
+          zk_save
+        rescue *ZK_CONNETION_ERRORS => e
+          log.error "nerve: error in reporting up on zk node #{@full_key}: #{e.message}"
+        end
+      end
     end
 
     def report_down
-      zk_delete
+      if not @zk.connected?
+        log.error "nerve: error in reporting down on zk node #{@full_key}: loss connection"
+      else
+        begin
+          zk_delete
+        rescue *ZK_CONNETION_ERRORS => e
+          log.error "nerve: error in reporting down on zk node #{@full_key}: #{e.message}"
+        end
+      end
     end
 
     def ping?
-      return @zk.connected? && @zk.exists?(@full_key || '/')
```

Comment:
it is needed to change this into two separate checks. what I observed in the test is that, when packet is blocked, there is a delay (a few sec) til the @zk.connected? returns false. So connection error can still be raised on the @zk.exists? check.

Reply:
👍